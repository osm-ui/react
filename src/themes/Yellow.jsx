import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from 'styled-components';
import _ from 'lodash';
import colors from 'themes/colors';
import { buildLightThemeConfig } from 'helpers/themes';
import { config as defaultThemeConfig, StyledSpan } from './Default';

export const config = _.merge(
  {},
  buildLightThemeConfig(defaultThemeConfig, colors, 'yellow'),
  {}
);

const YellowTheme = ({ children, ...rest }) => (
  <ThemeProvider theme={config}>
    <StyledSpan>
      {React.Children.map(children, child => {
        if (React.isValidElement(child)) {
          return React.cloneElement(child, rest);
        }

        return child;
      })}
    </StyledSpan>
  </ThemeProvider>
);

YellowTheme.propTypes = {
  children: PropTypes.node.isRequired
};

YellowTheme.displayName = 'YellowTheme';

export default YellowTheme;
