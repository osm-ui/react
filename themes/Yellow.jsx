
import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from 'styled-components';
import _ from 'lodash';
import colors from '../colors';
import { buildLightThemeConfig } from '../helpers/themes';
import {
    config as defaultThemeConfig,
    StyledDiv,
} from './Default';


export const config = _.merge(
    {},
    buildLightThemeConfig(defaultThemeConfig, colors, 'yellow'),
    {},
);


const YellowTheme = ({ children, ...rest }) => (
    <ThemeProvider theme={config}>
        <StyledDiv {...rest}>
            {children}
        </StyledDiv>
    </ThemeProvider>
);


YellowTheme.propTypes = {
    children: PropTypes.node.isRequired,
};

YellowTheme.displayName = 'YellowTheme';

export default YellowTheme;
