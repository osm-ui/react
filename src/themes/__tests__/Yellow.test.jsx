import React from 'react';
import {
  snapshotWithElementChildren,
  snapshotWithTextChildren
} from 'helpers/tests';
import Yellow from 'themes/Yellow';

describe('When using snapshots', () => {
  it('Should render with an element children', () =>
    snapshotWithElementChildren(Yellow));

  it('Should render with a text children', () =>
    snapshotWithTextChildren(Yellow));
});
