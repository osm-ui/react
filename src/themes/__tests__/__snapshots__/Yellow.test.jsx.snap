// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When using snapshots Should render with a text children 1`] = `
<styled.span>
  <YellowTheme
    key=".0"
  >
    This is a text
  </YellowTheme>
</styled.span>
`;

exports[`When using snapshots Should render with an element children 1`] = `
<styled.span>
  <YellowTheme
    key=".0"
  >
    <div />
  </YellowTheme>
</styled.span>
`;
